modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -317,6 +334,60 @@ public CompletableFuture<ExecutionTarget> forSystemView(ExecutionTargetFactory f
         services.forEach(LifecycleAware::start);
     }
 
+    /** Get primary replicas. */
+    private CompletableFuture<List<PrimaryReplica>> primaryReplicas(int tableId) {
```

Comment:
Could you please clarify when given method is actually used?
Let's say that we have following flow:
```
tx1 = transactions.begin();
insert into t1 values(k1,v1);
insert into t1 values(k2, v2);
tx1.commit();
```
When will the primaryReplcias() method be called? How many times? How do you manage previously retrieved primary replicas? Especially when it was changes (hope that you won't silently overwrite it in the inner primary replicas cache)? 

What I mean here, is that from transactions point of view it's only required to await primary replicas on **first** only.

Reply:
this code will be called on request mapping phase, 2 times (for now). There is no additional nodes caching are present now. And seems described case is not  case for this issue, i fill additional comment here https://github.com/apache/ignite-3/pull/2846/files#r1399126739

Reply:
Well, I'm not sure whether this should be fixed within given issue or another one. The main point is that, there's no sense to awaitPrimary on same partitions touch or it may even lead to inappropriate group re-enlistment.